src/blockdata/block.rs

```diff
@@ -165,15 +165,15 @@ impl BlockHeader {
 
     /// Checks that the proof-of-work for the block is valid.
     pub fn validate_pow(&self, required_target: &Uint256) -> Result<(), util::Error> {
-        use byteorder::{ByteOrder, LittleEndian};
-
         let target = &self.target();
         if target != required_target {
             return Err(BlockBadTarget);
         }
         let data: [u8; 32] = self.bitcoin_hash().into_inner();
         let mut ret = [0u64; 4];
-        LittleEndian::read_u64_into(&data, &mut ret);
+        for (ret_val, data_bytes) in ret.iter_mut().zip(data.chunks(4)) {
```

Comment:
If we're gonna replace a crate like byteorder which is obviously meant to avoid silly programming errors by providing utility functions, I feel like we should have a method in endian.rs for every byteorder method we are replacing. Instead of inlining parts and having a method for others.

If I want to go do `LittleEndian::read_u64_into` in another section of code in the future in rust-bitcoin, I shoulnd't have to think about this again.

Reply:
Changed it into a macro so is pretty hard to screw up now.

Reply:
I don't see any macro here :)
An it should be more performant to use ` chunks_exact` if the iterator is dividable exactly.

Reply:
Sadly chunks_exact was added in 1.31. fn'd.